### DISCUSSION TITLE
Help connecting Artisan (Mac) to Toper 1kg

### DISCUSSION THREAD

**Discussion by dlehman** [artisan-roaster-scope/artisan#1151] [General]
I have a Toper TKMSX 1kg roaster. I would like to connect it to Artisan v2.8.2 running on a MacBook Pro (M1, Ventura). The roaster has a USB cable attached to a Phidget 1048_0, but no network (Ethernet) port in the roaster, that I can find.

So far, I have not been able to get the roaster connected to Artisan (ie. showing any temps).

Initially, I tried using the "Config > Machine > Toper > USB" menu in Artisan. However, it then asks me to chose a COM port, which is a Windows-only convention (Mac names serial ports differently). I tried to figure out what the `/dev/tty` name of the Toper USB serial device was, but it doesn't show as a regular USB device.

As per a StackOverflow post, I can run an ioreg command and see a `PhidgetTemperatureSensor@00100000` device show up as a "IOUSBHostDevice" but I don't know if that helps me at all. It does not show up as a `/dev/ttyusb*` device (or any /dev/tty device, that I can see).

Then I figured out the USB cable is attached to a Phidget module inside the roaster (PhidgetTemperatureSensor 4-Input 1048_0). So I installed the Mac Phidget Control Panel from phidgets.com. The Phidget device shows up when I plug in the USB cable from the roaster. In Artisan, I tried "Config > Machine > Phidget > USB 1048 Databridge 2xTC". When I run Artisan, it gives an error:

"Exception: PhidgetManager couldn't be started. Verify that the Phidget driver is correctly installed!"

So obviously I have no idea what i'm doing. Can someone give me a shove in the right direction?

1. Should Artisan work using the "Machine>Toper>USB" config, or should I use the "Machine>Phidget>USB..." path instead?

2. On an M1 Mac running Ventura (macOS 13), it blocks System Extensions by default. Can I use the Phidget Control Panel _instead_ of installing extra drivers? There is a note on the Phidgets site that says "If you're having issues with the regular installer, try installing and running the Standalone Control Panel, which uses a driver extension instead of a kernel extension." Or should I force the kext (kernel extension) to install?

Any advice would be appreciated.

**Comment by MAKOMO** [comment]
Please don't crosspost and disturb other threads here!

You have obviously a Phidget installed in your machine. Artisan can only talk to that Phidget if the Phidget driver is correctly installed, which it is not in your case. Allow the regular Phidget installer to install its kernel extension. The Standalone Control Panel is of no use in combination with Artisan. Once this driver is correctly installed, verify that it works by opening the Phidget Control Panel app which comes along that Phidget installer. If that shows your module and receives the expected temperatures, close it and configure Artisan for this Phidget (menu Machine >> Phidget >> USB 1048...).

**Comment by dlehman** [comment]
Thank-you for your quick reply, and clear instructions. Appreciate you taking the time to set me straight.
